community/detectors/localai_cve_2024_2029/src/main/java/com/google/tsunami/plugins/detectors/rce/cve20242029/LocalAiCve20242029RceDetector.java

```diff
@@ -0,0 +1,237 @@
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final PayloadGenerator payloadGenerator;
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+  private final int oobSleepDuration;
+
+  @Inject
+  LocalAiCve20242029RceDetector(
+      HttpClient httpClient,
+      @UtcClock Clock utcClock,
+      PayloadGenerator payloadGenerator,
+      @OobSleepDuration int oobSleepDuration) {
+    this.httpClient = checkNotNull(httpClient);
+    this.utcClock = checkNotNull(utcClock);
+    this.payloadGenerator = checkNotNull(payloadGenerator);
+    this.oobSleepDuration = oobSleepDuration;
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("Starting LocalAI CVE-2024-2029 RCE detection.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::checkIfLocalAiWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean checkIfLocalAiWebService(NetworkService networkService) {
+    String targetUrl = NetworkServiceUtils.buildWebApplicationRootUrl(networkService);
+
+    String modelsUrl = targetUrl + "models";
+    HttpResponse response;
+    try {
+      response = httpClient.send(get(modelsUrl).withEmptyHeaders().build(), networkService);
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Unable to query '%s'.", targetUrl);
+      return false;
+    }
+    if (response.bodyJson().isEmpty()) {
+      return false;
+    }
+    try {
+      return !response.bodyJson().get().getAsJsonObject().get("object").getAsString().isEmpty();
+    } catch (IllegalStateException e) {
+      logger.atWarning().withCause(e).log("Unable to get response body as a JSON Object.");
+    }
+    return false;
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    var payload =
+        this.payloadGenerator.generate(
+            PayloadGeneratorConfig.newBuilder()
+                .setVulnerabilityType(PayloadGeneratorConfig.VulnerabilityType.REFLECTIVE_RCE)
+                .setInterpretationEnvironment(
+                    PayloadGeneratorConfig.InterpretationEnvironment.LINUX_SHELL)
+                .setExecutionEnvironment(
+                    PayloadGeneratorConfig.ExecutionEnvironment.EXEC_INTERPRETATION_ENVIRONMENT)
+                .build());
+    if (payload == null || !payload.getPayloadAttributes().getUsesCallbackServer()) {
+      logger.atWarning().log(
+          "The Tsunami callback server is not setup for this environment, so we cannot confirm the"
+              + " RCE callback");
+      return false;
+    }
```

Comment:
Thanks @devampkid!
This check is not needed anymore, please remove it, then we should be good to go.